streams/src/main/java/org/apache/kafka/streams/state/internals/SegmentIterator.java

```diff
@@ -67,14 +70,22 @@ public Bytes peekNextKey() {
     public boolean hasNext() {
         boolean hasNext = false;
         while ((currentIterator == null || !(hasNext = hasNextConditionHasNext()) || !currentSegment.isOpen())
-                && segments.hasNext()) {
+            && segments.hasNext()) {
             close();
             currentSegment = segments.next();
             try {
                 if (from == null || to == null) {
-                    currentIterator = currentSegment.all();
+                    if (forward) {
+                        currentIterator = currentSegment.all();
+                    } else {
+                        currentIterator = currentSegment.reverseAll();
```

Comment:
This is not directly related to this PR, but it makes me wondering: why do not keep a separate `range / all` in extended `Segment` interface? Should we just remove that? Now we've added the reverse ones but only in the parent interface, it makes me feeling the original ones on `Segment` is not necessary.

Reply:
Yeah that's a good question, it does seem like we can just remove the `range` and `all` on the Segment interface

Reply:
+1. Applying change.